### DISCUSSION TITLE
Refactoring and next version of OIDC

### DISCUSSION THREAD

**Discussion by muhlemmer** [zitadel/oidc#276] [Ideas]
Dear community. I've recently started at zitadel. Firstly I'm tasked to help evolve this OIDC library. I will look in things like  #116, merging of #173 and refactoring the library to use more concrete typing instead of interfaces for return types. Where possible or needed, we are also thinking about using generics to allow consumers to replace certain types for customization.

All of the above ideas will imply breaking changes, so we are planning a new major release once all of that is done. In order to facilitate this and keep `main` stable, we have created the `next` branch where breaking change pull-requests will be targeted. Also, issues and pull-request targeting the next version will be added to the [v2 milestone](https://github.com/zitadel/oidc/milestone/1), so we can keep track of them.

@muir, I've noticed the `NEXT_RELEASE.md` file. If you have some time to spare, may I ask you to convert these lines into issues, so that we can pick them up? Or pull requests are also welcome, of course :)

https://github.com/zitadel/oidc/blob/3a6c3543e7b8acc00d666420cdfe684c3ddd0bf2/NEXT_RELEASE.md?plain=1#L4-L6

**Comment by muhlemmer** [comment]
WIP: #283

**Comment by muhlemmer** [comment]
@zitadel/engineers should we also get rid of the `op.Configuration` interface? It is currently implemented only by `op.Provider`. `op.Provider` is configured by the exported `op.Config` , which is a subset of the options in `op.Configuration` .

https://github.com/zitadel/oidc/blob/c8d61c0858de6b503e38afd8798437b916044963/pkg/op/config.go#L17-L43

**Reply by adlerhurst** [reply]
Sounds good to me. I think i haven't used any other lib where config is an interface

**Comment by muir** [comment]
@muhlemmer 
The things I put in NEXT-RELEASE.md are all trivial changes.  The reason I didn't already do them is that they break backwards compatibility.  I can send a PR if you want.

**Reply by muhlemmer** [reply]
That would be great. Please target the `next` branch for this.

**Comment by muir** [comment]
I also want to draw your attention to a particular discord thread.  It's not active right now, but it's an appropriate place for discussions like these: https://discord.com/channels/927474939156643850/1019895518383583323/threads/1024593088016031744

**Reply by muhlemmer** [reply]
The link doesn't drop me in  any thread somehow. Just the server. That might be lack of integration between my browser and the discord app on gentoo :P. Would you mind making a mention in the thread? My nick is `@tim` on discord.
